### DISCUSSION TITLE
Disable public signup

### DISCUSSION THREAD

**Discussion by BackedUpBooty** [ghostfolio/ghostfolio#970] [Ideas]
Hey there, following your request on Slack for me to create a ticket here:

It would be great if there were a variable to limit or restrict new user sign ups. Something like `DISABLE_NEW_USERS=` with `false` as default but then allowing us to set it as `true` in the future. If I knew more about programming I'd have happily tried to come up with something myself to share but alas, I'm not actually any good at it!

**Comment by dtslvr** [comment]
You can fund the development of this feature via the crowdfunding campaign [here](https://www.buymeacoffee.com/ghostfolio/w/5111).

**Reply by gobdevel** [reply]
I can work on this feature, please let me know all requirements related to this.

**Reply by dtslvr** [reply]
Hi @gobdevel

Thanks for your interest. Here is a short solution sketch.

### Backend
* Introduce a new `globalPermission` for `ENABLE_USER_SIGNUP` here: https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/info/info.service.ts#L54
* Protect the endpoint accordingly: https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/user/user.controller.ts#L70

### Frontend
* Add a `mat-slide-toggle` with a label _Enable User Signup_ in `localhost:4200/en/admin/overview` (cf. https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/components/admin-overview/admin-overview.html#L112)
* You can reuse `putAdminSetting()` which sets the flag in the backend (Table `Property`)
* Conditionally show the _Get Started_ button here https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/components/header/header.component.html#L291 and https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/landing/landing-page.html#L34

Just reach out if you have any questions.

**Reply by gobdevel** [reply]
PR has been raised for this feature -
[PR-1512](https://github.com/ghostfolio/ghostfolio/pull/1512)

Please have a look.

**Reply by dtslvr** [reply]
Thank you very much for your contribution @gobdevel! 😎 

I have refactored your code a bit. Please take a look if you are interested.

**Reply by gobdevel** [reply]
Thanks @dtslvr for taking this forward...
I see you have removed code which were blocking user registration from third party (Internet-identity & O-auth). What is the idea behind this ?
With current code committed, I believe backend still allow user registration via third party signup.

**Reply by dtslvr** [reply]
Yes, you are right. I removed it first step because there is no official support for social login.

Readded here: #1515

**Reply by gobdevel** [reply]
Thank you.. now working this feature fully.. verified.

**Reply by dtslvr** [reply]
🙌🏻 

By the way, very impressive findings for a first-time contributor!

**Comment by BackedUpBooty** [comment]
Thank you both, this will be really helpful!

**Comment by dtslvr** [comment]
This feature has landed in Ghostfolio [1.219.0](https://github.com/ghostfolio/ghostfolio/releases/tag/1.219.0).
